src/main/java/de/rwth/idsg/steve/web/controller/WebUsersController.java

```diff
@@ -0,0 +1,170 @@
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import jakarta.validation.Valid;
+
+
+@Controller
+@RequestMapping(value = "/manager/webusers")
+public class WebUsersController {
+
+    @Autowired private WebUserService webUserService;
+
+    private static final String PARAMS = "params";
+
+    // -------------------------------------------------------------------------
+    // Paths
+    // -------------------------------------------------------------------------
+
+    private static final String QUERY_PATH = "/query";
+
+    private static final String DETAILS_PATH = "/details/{webuserpk}";
+    private static final String DELETE_ALL_PATH = "/delete/{webuserpk}";
+    private static final String UPDATE_PATH = "/update";
+    private static final String ADD_PATH = "/add";
+
+    // -------------------------------------------------------------------------
+    // HTTP methods
+    // -------------------------------------------------------------------------
+
+    @RequestMapping(method = RequestMethod.GET)
+    public String getOverview(Model model) {
+        initList(model, new WebUserQueryForm());
+        return "data-man/webusers";
+    }
+
+    @RequestMapping(value = QUERY_PATH, method = RequestMethod.GET)
+    public String getQuery(@ModelAttribute(PARAMS) WebUserQueryForm params, Model model) {
+        initList(model, params);
+        return "data-man/webusers";
+    }
+
+    private void initList(Model model, WebUserQueryForm params) {
+        model.addAttribute(PARAMS, params);
+        model.addAttribute("webuserList", webUserService.getOverview(params));
+    }
+
+    @RequestMapping(value = DETAILS_PATH, method = RequestMethod.GET)
+    public String getDetails(@PathVariable("webuserpk") Integer webuserpk, Model model) {
+       WebUserForm form = webUserService.getDetails(webuserpk);
+
+        model.addAttribute("webuserForm", form);
+        return "data-man/webuserDetails";
+    }
+
+    @RequestMapping(value = ADD_PATH, method = RequestMethod.GET)
+    public String addGet(Model model) {
+        model.addAttribute("webuserForm", new WebUserForm());
+        return "data-man/webuserAdd";
+    }
+
+    @RequestMapping(params = "add", value = ADD_PATH, method = RequestMethod.POST)
+    public String addPost(@Valid @ModelAttribute("webuserForm") WebUserForm webuserForm,
+                          BindingResult result, Model model) {
+        if (result.hasErrors()) {
+            return "data-man/webuserAdd";
+        }
+
+        // password is Null, Blank/Empty or less than 8 Characters then don't add and show an Error
+        if (webuserForm.getPassword() == null) {
+            webuserForm.setPwerror(Boolean.TRUE);
+            return "data-man/webuserAdd";
```

Comment:
this type of validation should be put into WebUserForm with validator annotations, and not into java code as plainly as you did. see other Form usages.

Reply:
Your suggestion collides at the moment with the update path, were we may or may not update the password.

Reply:
If you remove it from the form, it should fix most of the issues.
The password could be provided as result of the user creation.